internal/runtime/internal/controller/value_cache.go

```diff
@@ -1,88 +1,93 @@
 package controller
 
 import (
+	"fmt"
 	"reflect"
 	"sync"
 
 	"github.com/grafana/alloy/internal/component"
 	"github.com/grafana/alloy/syntax/vm"
 )
 
-// valueCache caches component arguments and exports to expose as variables for
-// Alloy expressions.
-//
-// The current state of valueCache can then be built into a *vm.Scope for other
-// components to be evaluated.
+// This special keyword is used to expose the argument values to the custom components.
+const argumentLabel = "argument"
```

Comment:
Would it be cleaner to just have another map for arguments?

Reply:
yeah I agree, I wanted to have the Scope fully built so that when a node wants to evaluate it can just get a copy of it but I don't think that's worth the added complexity in GetModuleArgument, CacheModuleArgument and SyncModuleArgs